### PR TITLE
Adds .travis.yml

### LINKED ISSUE #23: Hook up to CI
It looks like pulumi/pulumi is using Travis for CI, so I assume we'll want to do the same here?

Obviously a full suite of unit tests would be nice, but maybe we should start with just a quick smoke check to make sure things build properly?

### PR DESCRIPTION
**Pull request by liftedkilt**
This adds at least initial support for Travis. Closes #23 (probably?)

### CODE REVIEW

**Comment by hausdorff** [comment]
Ha, beat me to it. This was on my list for today. :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: go
+go:
+ - "1.11.x"
+
+env:
+  - GO111MODULE=on
```
Comment (hausdorff) [inline .travis.yml]: Tiny nit, might want to add a newline here, otherwise git complains about it in the diffs.

**Comment by hausdorff** [comment]
BTW, @liftedkilt, I feel bad asking you to do this work by yourself -- if you want to pair-debug this, I'm @hausdorff on the Kubernetes Slack, and also @hausdorff on [the Pulumi slack](https://slack.pulumi.io/). Feel free to bother me any time about this.

**Comment by hausdorff** [comment]
@liftedkilt since I'm going to be working on the `Makefile` stuff anyway, I will also try to take a whack at fixing this... if I succeed before you do feel free to absorb the change into your PR and we'll commit under your name. :)

**Comment by hausdorff** [comment]
@liftedkilt ok, here's a `go.mod` file that works. I have no idea if I am doing this right, but if I am, I anticipate this will continue to be a serious amount of work to maintain in the future.

```
module github.com/pulumi/kubespy

require (
	cloud.google.com/go v0.28.0 // indirect
	github.com/Azure/go-autorest v9.4.0+incompatible // indirect
	github.com/NYTimes/gziphandler v1.0.1 // indirect
	github.com/PuerkitoBio/purell v1.1.0 // indirect
	github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
	github.com/alcortesm/tgz v0.0.0-20161220082320-9c5fe88206d7 // indirect
	github.com/anmitsu/go-shlex v0.0.0-20161002113705-648efa622239 // indirect
	github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973 // indirect
	github.com/bgentry/speakeasy v0.1.0 // indirect
	github.com/blang/semver v3.5.1+incompatible // indirect
	github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd // indirect
	github.com/coreos/bbolt v1.3.0 // indirect
	github.com/coreos/etcd v3.3.9+incompatible // indirect
	github.com/coreos/go-oidc v2.0.0+incompatible // indirect
	github.com/coreos/go-semver v0.2.0 // indirect
	github.com/coreos/go-systemd v0.0.0-20180828140353-eee3db372b31 // indirect
	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f // indirect
	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
	github.com/djherbis/times v1.0.1 // indirect
	github.com/docker/distribution v2.6.2+incompatible // indirect
	github.com/docker/spdystream v0.0.0-20170912183627-bc6354cbbc29 // indirect
	github.com/elazarl/go-bindata-assetfs v1.0.0 // indirect
	github.com/emicklei/go-restful v2.8.0+incompatible // indirect
	github.com/emicklei/go-restful-swagger12 v0.0.0-20170208215640-dcef7f557305 // indirect
	github.com/emirpasic/gods v1.9.0 // indirect
	github.com/evanphx/json-patch v4.1.0+incompatible // indirect
	github.com/fatih/color v1.7.0
	github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568 // indirect
	github.com/ghodss/yaml v0.0.0-20150909031657-73d445a93680 // indirect
	github.com/gliderlabs/ssh v0.1.1 // indirect
	github.com/go-openapi/jsonpointer v0.0.0-20180322222829-3a0015ad55fa // indirect
	github.com/go-openapi/jsonreference v0.0.0-20180322222742-3fb327e6747d // indirect
	github.com/go-openapi/spec v0.0.0-20180801175345-384415f06ee2 // indirect
	github.com/go-openapi/swag v0.0.0-20180715190254-becd2f08beaf // indirect
	github.com/gogo/protobuf v0.0.0-20170330071051-c0656edd0d9e // indirect
	github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4 // indirect
	github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c // indirect
	github.com/google/go-cmp v0.2.0 // indirect
	github.com/google/gofuzz v0.0.0-20161122191042-44d81051d367 // indirect
	github.com/googleapis/gnostic v0.2.0 // indirect
	github.com/gophercloud/gophercloud v0.0.0-20180922034555-725b2c0b2427 // indirect
	github.com/gorilla/websocket v1.4.0 // indirect
	github.com/gosuri/uilive v0.0.0-20170323041506-ac356e6e42cd
	github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7 // indirect
	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0 // indirect
	github.com/grpc-ecosystem/grpc-gateway v1.5.0 // indirect
	github.com/hashicorp/go-multierror v1.0.0 // indirect
	github.com/hashicorp/golang-lru v0.5.0 // indirect
	github.com/howeyc/gopass v0.0.0-20170109162249-bf9dde6d0d2c // indirect
	github.com/imdario/mergo v0.3.6 // indirect
	github.com/inconshreveable/mousetrap v1.0.0 // indirect
	github.com/jbenet/go-context v0.0.0-20150711004518-d14ea06fba99 // indirect
	github.com/jonboulle/clockwork v0.1.0 // indirect
	github.com/json-iterator/go v1.1.5 // indirect
	github.com/k0kubun/colorstring v0.0.0-20150214042306-9440f1994b88 // indirect
	github.com/kevinburke/ssh_config v0.0.0-20180830205328-81db2a75821e // indirect
	github.com/mailru/easyjson v0.0.0-20180823135443-60711f1a8329 // indirect
	github.com/mattn/go-colorable v0.0.9 // indirect
	github.com/mattn/go-isatty v0.0.4 // indirect
	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
	github.com/mitchellh/go-homedir v1.0.0 // indirect
	github.com/mitchellh/go-ps v0.0.0-20170309133038-4fdf99ab2936 // indirect
	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
	github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742 // indirect
	github.com/mxk/go-flowrate v0.0.0-20140419014527-cca7078d478f // indirect
	github.com/onsi/gomega v1.4.2 // indirect
	github.com/opentracing/opentracing-go v1.0.2 // indirect
	github.com/pborman/uuid v1.2.0 // indirect
	github.com/pelletier/go-buffruneio v0.2.0 // indirect
	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
	github.com/pkg/errors v0.8.0 // indirect
	github.com/pquerna/cachecontrol v0.0.0-20180517163645-1555304b9b35 // indirect
	github.com/prometheus/client_golang v0.8.0 // indirect
	github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910 // indirect
	github.com/prometheus/common v0.0.0-20180801064454-c7de2306084e // indirect
	github.com/prometheus/procfs v0.0.0-20180920065004-418d78d0b9a7 // indirect
	github.com/pulumi/pulumi v0.15.4 // indirect
	github.com/pulumi/pulumi-kubernetes v0.17.0
	github.com/reconquest/loreley v0.0.0-20160708080500-2ab6b7470a54 // indirect
	github.com/sergi/go-diff v1.0.0 // indirect
	github.com/sirupsen/logrus v1.1.0 // indirect
	github.com/soheilhy/cmux v0.1.4 // indirect
	github.com/spf13/cobra v0.0.3
	github.com/spf13/pflag v1.0.2 // indirect
	github.com/src-d/gcfg v1.3.0 // indirect
	github.com/stevvooe/resumable v0.0.0-20180830230917-22b14a53ba50 // indirect
	github.com/texttheater/golang-levenshtein v0.0.0-20180516184445-d188e65d659e // indirect
	github.com/tmc/grpc-websocket-proxy v0.0.0-20171017195756-830351dc03c6 // indirect
	github.com/uber/jaeger-client-go v2.14.0+incompatible // indirect
	github.com/uber/jaeger-lib v1.5.0 // indirect
	github.com/ugorji/go/codec v0.0.0-20180927125128-99ea80c8b19a // indirect
	github.com/xanzy/ssh-agent v0.2.0 // indirect
	github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18 // indirect
	github.com/yudai/gojsondiff v0.0.0-20170107030110-7b1b7adf999d
	github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82 // indirect
	github.com/yudai/pp v2.0.1+incompatible // indirect
	golang.org/x/time v0.0.0-20161028155119-f51c12702a4d // indirect
	google.golang.org/appengine v1.2.0 // indirect
	google.golang.org/grpc v1.15.0 // indirect
	gopkg.in/inf.v0 v0.9.0 // indirect
	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
	gopkg.in/square/go-jose.v2 v2.1.9 // indirect
	gopkg.in/src-d/go-billy.v4 v4.2.1 // indirect
	gopkg.in/src-d/go-git-fixtures.v3 v3.1.1 // indirect
	gopkg.in/src-d/go-git.v4 v4.7.0 // indirect
	gopkg.in/warnings.v0 v0.1.2 // indirect
	k8s.io/api v0.0.0-20180925152912-a191abe0b71e // indirect
	k8s.io/apiextensions-apiserver v0.0.0-20180925155151-c0b566b8903b // indirect
	k8s.io/apimachinery v0.0.0-20180228050457-302974c03f7e
	k8s.io/apiserver v0.0.0-20180426121757-0841753fc26e // indirect
	k8s.io/client-go v7.0.0+incompatible
	k8s.io/kube-openapi v0.0.0-20180731170545-e3762e86a74c // indirect
	k8s.io/kubernetes v1.10.2 // indirect
	k8s.io/utils v0.0.0-20180918230422-cd34563cd63c // indirect
)

```

**Comment by hausdorff** [comment]
After testing against this PR, I've merged those `go.mod` changes.

**Comment by liftedkilt** [comment]
Sweet! I just rebased on top of those changes. Yeah dependency management when dealing with kubernetes libraries seems to be pretty much awful no matter which way you slice it.

**Comment by hausdorff** [comment]
@liftedkilt oh btw... I just merged a bunch of `Makefile` changes and stuff. Should make release management easier.

**Comment by hausdorff** [comment]
@liftedkilt let me know when you're ready ... I'm not sure if it's "done" yet.

**Comment by liftedkilt** [comment]
I think we can merge this. It'll at least give us a start. 👍🏻

**Pull request merged**